components/constellation/constellation.rs

```diff
@@ -2711,8 +2713,13 @@ where
 
         if let Some(webgl_threads) = self.webgl_threads.as_ref() {
             debug!("Exiting WebGL thread.");
-            if let Err(e) = webgl_threads.exit() {
-                warn!("Exit WebGL Thread failed ({})", e);
+            match webgl_threads.exit() {
```

Comment:
You need to pass the sender here, and then send the reply when the exit message is handled in the webgl thread. 

See for example https://github.com/servo/servo/pull/32897/files#diff-95aa2eb890fe794918afdadfb6a89e026ec80222ceb8e5a21768d528e9ce036cR2698